### PR TITLE
Adds a fancy new README file

### PR DESCRIPTION
**Pull request by Omniroot**
After discussions with the team, I decided I would try my hand at a "nice" `README` file that may be good to turn into a template in the future.

## Inspiration
- https://github.com/matiassingers/awesome-readme
- https://medium.com/@meakaakka/a-beginners-guide-to-writing-a-kickass-readme-7ac01da88ab3

## Using an HTML header
I decided to go with an HTML header, because I noticed that the README files I liked in the `awesome-readme` repo all did this.

### Centred
I also liked the centred design better than a left-aligned version.

**Note** this is officially a markdown anti-pattern (as is any formatting); however, I don't think that's a big deal for GFM, since it's not "official" markdown. Also, seeing as how many READMEs are using this pattern (at least for the header), I don't see a problem with it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,21 +1,61 @@
...
+  </a>
+  <a href="https://badge.fury.io/js/%40shopify%2Fjavascript-utilities">
+    <img src="https://badge.fury.io/js/%40shopify%2Fjavascript-utilities.svg" alt="npm version">
+  </a>
+  <a href="https://github.com/Shopify/javascript-utilities/blob/master/LICENSE.md">
+    <img src="https://img.shields.io/badge/License-MIT-green.svg" alt="License: MIT">
+  </a>
+</p>
+
+<div align="center">
```
Comment (ismail-syed) [inline README.md]: Nit picking and opinionated a bit. Feel free to argue against this. 

I prefer this: 
```html
<div align="center">
    <a href="https://shopify.com/">Website</a>
    <span> | </span>
    <a href="https://polaris.shopify.com/">Styleguide</a>
    <span> | </span>
    <a href="http://github.com/Shopify/javascript-utilities/blob/master/CHANGELOG.md">Changelog</a>
</div>
```

Comment (Omniroot) [inline README.md]: I'm indifferent, tbh, although I should become opinionated.

My thoughts with the current styling are that it helps, in general, to spot and fix typos when the text is on its own line.

File: README.md [hunk 2]
```diff
@@ -1,21 +1,61 @@
-# javascript-utilities
-
-[![CircleCI](https://circleci.com/gh/Shopify/javascript-utilities.svg?style=svg)](https://circleci.com/gh/Shopify/javascript-utilities)
-
-A set of utilities for JavaScript and TypeScript projects at Shopify.
+<h1 align="center">
+  <br>
+  <a href="http://www.shopify.com/"><img src="https://cdn.shopify.com/assets2/press/brand/shopify-logo-main-small-f029fcaf14649a054509f6790ce2ce94d1f1c037b4015b4f106c5a67ab033f5b.png" alt="Shopify" width="200" /></a>
```
Comment (lemonmade) [inline README.md]: Really don’t like the big logo at the top, if we want to include it I think it should be below the license.

File: README.md [hunk 3]
```diff
@@ -1,21 +1,61 @@
...
+  <br>
+  javascript-utilities
+  <br>
+</h1>
+
+<h4 align="center">A set of utilities for JavaScript and TypeScript projects at Shopify.</h4>
```
Comment (lemonmade) [inline README.md]: We can’t just toss a doc with styles to center around this whole thing?

File: README.md [hunk 4]
```diff
@@ -1,21 +1,61 @@
...
+    <img src="https://badge.fury.io/js/%40shopify%2Fjavascript-utilities.svg" alt="npm version">
+  </a>
+  <a href="https://github.com/Shopify/javascript-utilities/blob/master/LICENSE.md">
+    <img src="https://img.shields.io/badge/License-MIT-green.svg" alt="License: MIT">
+  </a>
+</p>
+
+<div align="center">
+    <a href="https://shopify.com/">
+      Website
```
Comment (lemonmade) [inline README.md]: Why are we linking to these things? This feels so counter to what I expect from an OSS project, we shouldn’t be hiding the “what is this and why is it good and how do I use it” behind a wall of branding IMO

File: README.md [hunk 5]
```diff
@@ -1,21 +1,61 @@
...
+    </a>
+    <span> | </span>
+    <a href="https://polaris.shopify.com/">
+      Styleguide
+    </a>
+    <span> | </span>
+    <a href="http://github.com/Shopify/javascript-utilities/blob/master/CHANGELOG.md">
```
Comment (lemonmade) [inline README.md]: This link is good but make it part of the TOC

**Comment by Omniroot** [comment]
@lemonmade, when you get back, can you take another look at this?

**Comment by Omniroot** [comment]
I'm going to go ahead and merge this in, since all comments have been addressed.

**Pull request merged**